scipy/optimize/tests/test_optimize.py

```diff
@@ -1188,15 +1188,15 @@ def dfunc(z):
 
         for method in ['nelder-mead', 'powell', 'cg', 'bfgs',
                        'newton-cg', 'l-bfgs-b', 'tnc',
-                       'cobyla', 'slsqp']:
-            if method in ('nelder-mead', 'powell', 'cobyla'):
+                       'cobyla', 'cobyqa', 'slsqp']:
+            if method in ('nelder-mead', 'powell', 'cobyla', 'cobyqa'):
                 jac = None
             else:
                 jac = dfunc
 
-            sol1 = optimize.minimize(func, [1, 1], jac=jac, tol=1e-10,
+            sol1 = optimize.minimize(func, [2, 2], jac=jac, tol=1e-10,
```

Comment:
I think there are several locations in this file where cobyqa needs to be added.

- Please add a `test_cobyqa` method to `CheckOptimizeParameterized`. Examine the other methods to see how they're setup.
- https://github.com/scipy/scipy/blob/main/scipy/optimize/tests/test_optimize.py#L1647
- https://github.com/scipy/scipy/blob/main/scipy/optimize/tests/test_optimize.py#L1318